rocketpy/mathutils/function.py

```diff
@@ -224,6 +230,8 @@ def set_source(self, source):  # pylint: disable=too-many-statements
         else:
             # Evaluate dimension
             self.__dom_dim__ = source.shape[1] - 1
+            self._domain = source[:, :-1]
+            self._image = source[:, -1]
```

Comment:
isn't these lines equivalent to `x_array` and `y_array`?

Reply:
No, the y_array is the second input column. Therefore, in 1-D cases y_array is the output. For N-D it is only a particular column of the input domain.